### PR TITLE
doc: initial draft of dev workflows section

### PR DESCRIPTION
**Pull request by mhdawson**
Signed-off-by: Michael Dawson <mdawson@devrus.com>

### CODE REVIEW

File: docs/development/dev-flows.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+Regardless of the development flow, developers typically develop on Windows or MacOS with Linux being
+distant third. When running on Windows the trend is towards the use of the Windows Subsystem for Linux.
+
+With developers developing on Windows/MacOS and the deployment target most often being Linux in
+a container, the team's experience is that it is important for the developer to do some level of
+testing in a container as part of their workflow.
+
+In the past the most common approach for building/running containers on Windows and MacOS
+was docker desktop. Now that docker desktop is longer free, development teams are exploring other
```
Comment (lholmquist) [inline docs/development/dev-flows.md:25]:
```suggestion
was docker desktop. Now that docker desktop is no longer free, development teams are exploring other

```

File: docs/development/dev-flows.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  familiar with Linux). Docker client is still free and it can connect to a remove virtual
+  machine.
+* podman and podman desktop which provides similar functionality to docker desktop but is
+  open source and free.
+
+## Fully local development, native platform
+
+Developers build/test in the local environment (Windows/MacOS) using npm start and once
+ready push to CI where the target container is built. Testing with other components
+is limited to test/integration environments. Any local testing uses mocs for other components.
```
Comment (lholmquist) [inline docs/development/dev-flows.md:37]:
```suggestion
is limited to test/integration environments. Any local testing uses mocks for other components.

```

File: docs/development/dev-flows.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  in the development process.
+* No need for network access in order while developing/testing 
+
+Disadvantages:
+* more complex to setup and manage for each developer
+* significant developer machine requirements to run kubernetes and other components
+* harder to manage configuration drift between development environment and
+  deployment environment
+
+## Local component development, remove services
```
Comment (lholmquist) [inline docs/development/dev-flows.md:77]:
```suggestion
## Local component development, remote services

```

File: docs/development/dev-flows.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+Developer build/test/integrate in containers and access other components for
+testing in a shared development/test environment. 
+
+Advantages:
+* More limited resource requirement for developer machines
+* Less management/complexity for developer
+
+Disadvantages:
+* Network connectivity is required to develop
+* Concurent use of of shared development/test environment running other
```
Comment (lholmquist) [inline docs/development/dev-flows.md:88]:
```suggestion
* Concurrent use of of shared development/test environment running other

```

File: docs/development/dev-flows.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
+﻿# Typical Development Workflows
+
+Developement flows within the team's developer and the customers the team works
+with need to accomodate the following:
+
+* the end target/deployment artifact is most often a container
```
Comment (mhdawson) [inline docs/development/dev-flows.md:6]: * for some platforms, deployment remains a server or virtual machine

**Review by joesepi — APPROVED** [review]
LGTM

**Review by jimlindeman — APPROVED** [review]

**Pull request merged**